### ISSUE TITLE
kube-vip pod not visible in kube-system namespace

### ISSUE THREAD

**Issue by grunlab** [plunder-app/kube-vip#145]
**Describe the bug**
kube-vip pod not visible in kube-system namespace

**To Reproduce**
1. Create /etc/kubenetes/manifests/kube-vip.yaml (see content below)
2. kubeadm init first control
3. OK, kube-vip is running well and VIP allocated to control plan
```
root@master-02:~# crictl pods
POD ID              CREATED             STATE               NAME                                NAMESPACE           ATTEMPT
811071d47db2a       32 minutes ago      Ready               coredns-f9fd979d6-htz8m             kube-system         0
6f3890e8878ae       32 minutes ago      Ready               coredns-f9fd979d6-wzg7l             kube-system         0
a0292f7d47e7b       33 minutes ago      Ready               weave-net-s29gr                     kube-system         0
aa3012aff4c8c       34 minutes ago      Ready               kube-proxy-qkldd                    kube-system         0
e0788d6e8fdd9       35 minutes ago      Ready               etcd-master-02                      kube-system         0
fe9782ab938fd       35 minutes ago      Ready               kube-scheduler-master-02            kube-system         0
-->66f2308e660a0       35 minutes ago      Ready               kube-vip-master-02                  kube-system         0
f3eab3f883ad7       35 minutes ago      Ready               kube-controller-manager-master-02   kube-system         0
ec4f29a688077       35 minutes ago      Ready               kube-apiserver-master-02            kube-system         0
```
```
root@master-02:~# ip a
1: lo: <LOOPBACK,UP,LOWER_UP> mtu 65536 qdisc noqueue state UNKNOWN group default qlen 1000
    link/loopback 00:00:00:00:00:00 brd 00:00:00:00:00:00
    inet 127.0.0.1/8 scope host lo
       valid_lft forever preferred_lft forever
    inet6 ::1/128 scope host 
       valid_lft forever preferred_lft forever
4: enx001e063262b1: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc pfifo_fast state UP group default qlen 1000
    link/ether 00:1e:06:32:62:b1 brd ff:ff:ff:ff:ff:ff
    inet 192.168.0.102/24 brd 192.168.0.255 scope global noprefixroute enx001e063262b1
       valid_lft forever preferred_lft forever
-->    inet 192.168.0.200/32 scope global enx001e063262b1
       valid_lft forever preferred_lft forever
    inet6 fe80::2202:4b8d:c16b:e777/64 scope link noprefixroute 
       valid_lft forever preferred_lft forever
...
```

4. First control plan ready
```
root@master-02:~# kubectl --kubeconfig /etc/kubernetes/admin.conf get node -o wide
NAME        STATUS   ROLES    AGE   VERSION   INTERNAL-IP     EXTERNAL-IP   OS-IMAGE                       KERNEL-VERSION     CONTAINER-RUNTIME
master-02   Ready    master   32m   v1.19.6   192.168.0.102   <none>        Debian GNU/Linux 10 (buster)   5.4.83-odroidxu4   cri-o://1.19.0
```
5. kube-vip-master-02 pod not visible in kube-system namespace ???
```
root@master-02:~# kubectl --kubeconfig /etc/kubernetes/admin.conf get pod -o wide -n kube-system
NAME                                READY   STATUS    RESTARTS   AGE   IP              NODE        NOMINATED NODE   READINESS GATES
coredns-f9fd979d6-htz8m             1/1     Running   0          33m   10.32.0.3       master-02   <none>           <none>
coredns-f9fd979d6-wzg7l             1/1     Running   0          33m   10.32.0.2       master-02   <none>           <none>
etcd-master-02                      1/1     Running   0          33m   192.168.0.102   master-02   <none>           <none>
kube-apiserver-master-02            1/1     Running   0          33m   192.168.0.102   master-02   <none>           <none>
kube-controller-manager-master-02   1/1     Running   0          33m   192.168.0.102   master-02   <none>           <none>
kube-proxy-qkldd                    1/1     Running   0          33m   192.168.0.102   master-02   <none>           <none>
kube-scheduler-master-02            1/1     Running   0          33m   192.168.0.102   master-02   <none>           <none>
weave-net-s29gr                     2/2     Running   0          32m   192.168.0.102   master-02   <none>           <none>
```

**Environment (please complete the following information):**
 - OS/Distro: Debian 10
 - Kubernetes Version: 1.19.6
 - Kube-vip Version: 0.3.0

**`Kube-vip.yaml`:**
```
root@master-02:~# cat /etc/kubernetes/manifests/kube-vip.yaml 
apiVersion: v1
kind: Pod
metadata:
  name: kube-vip
  namespace: kube-system
spec:
  containers:
  - args:
    - manager
    env:
    - name: vip_arp
      value: "true"
    - name: vip_interface
      value: enx001e063262b1
    - name: port
      value: "6443"
    - name: vip_cidr
      value: "32"
    - name: cp_enable
      value: "true"
    - name: cp_namespace
      value: kube-system
    - name: vip_leaderelection
      value: "true"
    - name: vip_leaseduration
      value: "5"
    - name: vip_renewdeadline
      value: "3"
    - name: vip_retryperiod
      value: "1"
    - name: vip_address
      value: 192.168.0.200
    image: plndr/kube-vip:0.3.0
    imagePullPolicy: Always
    name: kube-vip
    resources: {}
    securityContext:
      capabilities:
        add:
        - NET_ADMIN
        - SYS_TIME
    volumeMounts:
    - mountPath: /etc/kubernetes/admin.conf
      name: kubeconfig
  hostNetwork: true
  serviceAccountName: kube-vip
  volumes:
  - hostPath:
      path: /etc/kubernetes/admin.conf
    name: kubeconfig
```

**Comment by thebsdbox** [comment]
The "only" time I've seen this is if the namespace doesn't exist that the static pod is assigned to. I'm kind of at a loss, is there any mention of kube-vip in the kubelet logs?

**Comment by grunlab** [comment]
kubelet logs:
```
root@master-02:~# systemctl restart kubelet
root@master-02:~# journalctl -fu kubelet | grep kube-vip
déc. 22 16:05:51 master-02 kubelet[28660]: E1222 16:05:51.447029   28660 kubelet.go:1576] Failed creating a mirror pod for "kube-vip-master-02_kube-system(d60e70748a63973ecc543c9844c63b1b)": pods "kube-vip-master-02" is forbidden: a mirror pod may not reference service accounts
déc. 22 16:05:52 master-02 kubelet[28660]: E1222 16:05:52.445717   28660 kubelet.go:1576] Failed creating a mirror pod for "kube-vip-master-02_kube-system(d60e70748a63973ecc543c9844c63b1b)": pods "kube-vip-master-02" is forbidden: a mirror pod may not reference service accounts
```

Effectively, the serviceAccount "kupe-vip" (mentioned into /etc/kubernetes/manifests/kube-vip.yaml / generated with the command below) is not existing 'by default" in the namespace kube-system:

```
root@master-02:~# podman run --rm plndr/kube-vip:0.3.0 manifest pod --interface enx001e063262b1 --vip 192.168.0.200 --arp --leaderElection --controlplane
apiVersion: v1
kind: Pod
metadata:
  creationTimestamp: null
  name: kube-vip
  namespace: kube-system
spec:
  containers:
  - args:
    - manager
    env:
    - name: vip_arp
      value: "true"
    - name: vip_interface
      value: enx001e063262b1
    - name: port
      value: "6443"
    - name: vip_cidr
      value: "32"
    - name: cp_enable
      value: "true"
    - name: cp_namespace
      value: kube-system
    - name: vip_leaderelection
      value: "true"
    - name: vip_leaseduration
      value: "5"
    - name: vip_renewdeadline
      value: "3"
    - name: vip_retryperiod
      value: "1"
    - name: vip_address
      value: 192.168.0.200
    image: plndr/kube-vip:0.3.0
    imagePullPolicy: Always
    name: kube-vip
    resources: {}
    securityContext:
      capabilities:
        add:
        - NET_ADMIN
        - SYS_TIME
    volumeMounts:
    - mountPath: /etc/kubernetes/admin.conf
      name: kubeconfig
  hostNetwork: true
-->serviceAccountName: kube-vip
  volumes:
  - hostPath:
      path: /etc/kubernetes/admin.conf
    name: kubeconfig
status: {}
```
```
root@master-02:~# kubectl --kubeconfig /etc/kubernetes/admin.conf get sa -n kube-system | grep kube-vip
no result
```
Strangely, this was not preventing the pod to start at cri-o level but it was not visible at k8s level ...

So, I've removed the line "serviceAccountName: kube-vip" into /etc/kubernetes/manifests/kube-vip.yaml, restarted kubelet and it is now ok:
```
root@master-02:~# systemctl restart kubelet
root@master-02:~# kubectl --kubeconfig /etc/kubernetes/admin.conf get pod -o wide -n kube-system
NAME                                READY   STATUS    RESTARTS   AGE   IP              NODE        NOMINATED NODE   READINESS GATES
coredns-f9fd979d6-4dzfl             1/1     Running   0          76m   10.32.0.2       master-02   <none>           <none>
coredns-f9fd979d6-69m8w             0/1     Running   0          76m   10.32.0.3       master-02   <none>           <none>
etcd-master-02                      1/1     Running   0          76m   192.168.0.102   master-02   <none>           <none>
kube-apiserver-master-02            0/1     Running   0          76m   192.168.0.102   master-02   <none>           <none>
kube-controller-manager-master-02   0/1     Running   0          76m   192.168.0.102   master-02   <none>           <none>
kube-proxy-wjrcp                    1/1     Running   0          76m   192.168.0.102   master-02   <none>           <none>
kube-scheduler-master-02            0/1     Running   0          76m   192.168.0.102   master-02   <none>           <none>
--> kube-vip-master-02                  1/1     Running   0          6s    192.168.0.102   master-02   <none>           <none>
weave-net-fdvwg                     2/2     Running   0          51m   192.168.0.102   master-02   <none>           <none>
```

I think the line "serviceAccountName: kube-vip" should not be in the generated yaml when the option "--services" is not specified ... right ? In my case, i don't plan to use kube-vip to watch services.

**Comment by thebsdbox** [comment]
Ah, this is an excellent catch ! 

I'll have a think about options to mitigate this, the serviceAccount is required for a DaemonSet but Not the static pod.

**Comment by thebsdbox** [comment]
Fixed and version 0.3.0 released !
